HW02/Nastja_Kireeva/spec/array_module_spec.rb

```diff
@@ -0,0 +1,185 @@
+require_relative '../array_module'
+
+using ArrayModule
+
+RSpec.describe ArrayModule do
+  let(:empty_array) { [] }
+  let(:array) { [1, 2, 3, 4, 5, 6] }
+
+  describe '#my_empty?' do
+    context 'when array is not empty' do
+      it 'returns false' do
+        expect(array.my_empty?).to be false
+      end
+    end
+    context 'when array is empty' do
+      it 'returns true' do
+        expect(empty_array.my_empty?).to be true
+      end
+    end
+  end
+
+  describe '#my_map' do
+    let(:new_array) { [3, 6, 9, 12, 15, 18] }
+    context 'when block is given' do
+      context 'when array is empty' do
+        it 'returns an empty array' do
+          expect(empty_array.my_map(&:even?)).to be_empty
+        end
+      end
+      context 'when array is not empty' do
+        it 'returns a modified array' do
+          expect(array.my_map { |element| element * 3 }).to eq new_array
+        end
```

Comment:
If you use `new_array` only inside this context make sence to define it only here
```suggestion
      context 'when array is not empty' do
        let(:new_array) { [3, 6, 9, 12, 15, 18] }
        it 'returns a modified array' do
          expect(array.my_map { |element| element * 3 }).to eq new_array
        end
```